### PR TITLE
Add service wise extra controls and queries 

### PR DESCRIPTION
**Pull request by Priyanka-Chatterjee-2000**
### Checklist

1. [x] glue_data_catalog_encryption_enabled
2. [x] glue_security_configuration_encryption_enabled
3. [x] vpc_network_acl_allow_ftp_port_20_ingress
4. [x] vpc_network_acl_allow_ftp_port_21_ingress
5. [x] vpc_network_acl_allow_ssh_port_22_ingress
6. [x] vpc_network_acl_allow_rdp_port_3389_ingress
7. [x] glacier_vault_restrict_public_access
8. [x] sns_topic_policy_restrict_public_access
9. [x] vpc_network_acl_rule_restrict_ingress_ports_all

### CODE REVIEW

File: controls/glue.sp [hunk 1]
```diff
@@ -42,3 +44,21 @@ control "glue_job_security_configuration_enabled" {
 
   tags = local.glue_compliance_common_tags
 }
+
+
+control "glue_data_catalog_encryption_enabled" {
+  title       = "Glue data catalog encryption should be enabled"
+  description = "This control checks whether data catalog encryption is enabled. This control is non-complaint if data catalog encryption is disabled."
```
Comment (madhushreeray30) [inline controls/glue.sp:51]:
```suggestion
  description = "This control checks whether Glue data catalog encryption is enabled. This control is non-complaint if data catalog encryption is disabled."
```

File: controls/vpc.sp [hunk 2]
```diff
@@ -24,7 +29,7 @@ benchmark "vpc" {
     control.vpc_security_group_description_for_rules,
     control.vpc_security_group_rule_description_for_rules,
     control.vpc_subnet_auto_assign_public_ip_disabled,
-    control.vpc_transfer_server_not_publicly_accesible
+    control.vpc_transfer_server_not_publicly_accesible,
```
Comment (madhushreeray30) [inline controls/vpc.sp:32]:
```suggestion
    control.vpc_transfer_server_not_publicly_accesible
```

File: controls/vpc.sp [hunk 3]
```diff
@@ -192,3 +197,44 @@ control "vpc_ec2_transit_gateway_auto_accept_attachment_requests_disabled" {
 
   tags = local.vpc_compliance_common_tags
 }
+
+control "vpc_network_acl_allow_ftp_port_20_ingress" {
+  title       = "Network ACL should not allow unrestricted FTP port 20 access"
+  description = "This control checks whether the Network ACL allows restricted FTP port 20 ingress."
```
Comment (madhushreeray30) [inline controls/vpc.sp:203]:
```suggestion
  description = "This control checks whether the Network ACL allows unrestricted FTP port 20 ingress."
```

File: controls/vpc.sp [hunk 4]
```diff
@@ -192,3 +197,44 @@ control "vpc_ec2_transit_gateway_auto_accept_attachment_requests_disabled" {
...
+  query       = query.vpc_network_acl_allow_ftp_port_20_ingress
+
+  tags = local.vpc_compliance_common_tags
+}
+
+control "vpc_network_acl_allow_ftp_port_21_ingress" {
+  title       = "Network ACL should not allow unrestricted FTP port 21 access"
+  description = "This control checks whether the Network ACL allows restricted FTP port 21 ingress."
```
Comment (madhushreeray30) [inline controls/vpc.sp:211]:
```suggestion
  description = "This control checks whether the Network ACL allows unrestricted FTP port 21 ingress."
```

File: controls/vpc.sp [hunk 5]
```diff
@@ -192,3 +197,44 @@ control "vpc_ec2_transit_gateway_auto_accept_attachment_requests_disabled" {
...
+  query       = query.vpc_network_acl_allow_ftp_port_21_ingress
+
+  tags = local.vpc_compliance_common_tags
+}
+
+control "vpc_network_acl_allow_ssh_port_22_ingress" {
+  title       = "Network ACL should not allow unrestricted SSH port 22 access"
+  description = "This control checks whether the Network ACL allows restricted SSH port 22 ingress."
```
Comment (madhushreeray30) [inline controls/vpc.sp:219]:
```suggestion
  description = "This control checks whether the Network ACL allows unrestricted SSH port 22 ingress."
```

File: controls/vpc.sp [hunk 6]
```diff
@@ -192,3 +197,44 @@ control "vpc_ec2_transit_gateway_auto_accept_attachment_requests_disabled" {
...
+  query       = query.vpc_network_acl_allow_ssh_port_22_ingress
+
+  tags = local.vpc_compliance_common_tags
+}
+
+control "vpc_network_acl_allow_rdp_port_3389_ingress" {
+  title       = "Network ACL should not allow unrestricted RDP port 3389 access"
+  description = "This control checks whether the Network ACL allows restricted RDP port 3389 ingress."
```
Comment (madhushreeray30) [inline controls/vpc.sp:227]:
```suggestion
  description = "This control checks whether the Network ACL allows unrestricted RDP port 3389 ingress."
```

File: query/vpc.sp [hunk 7]
```diff
@@ -357,4 +357,202 @@ query "vpc_ec2_transit_gateway_auto_accept_attachment_requests_disabled" {
...
+    )
+    select
+      type || ' ' || r.name as resource,
+      case
+        when g.name is null then 'ok'
+        else 'alarm'
+      end as status,
+      r.name || case
+        when g.name is null then ' restricts FTP port 21 access from internet'
+        else ' allows FTP port 21 access from internet'
```
Comment (madhushreeray30) [inline query/vpc.sp:438]:
```suggestion
        else ' allows FTP port 21 access from the internet'
```

File: query/vpc.sp [hunk 8]
```diff
@@ -357,4 +357,202 @@ query "vpc_ec2_transit_gateway_auto_accept_attachment_requests_disabled" {
...
+        )
+    )
+    select
+      type || ' ' || r.name as resource,
+      case
+        when g.name is null then 'ok'
+        else 'alarm'
+      end as status,
+      r.name || case
+        when g.name is null then ' restricts FTP port 21 access from internet'
```
Comment (madhushreeray30) [inline query/vpc.sp:437]:
```suggestion
        when g.name is null then ' restricts FTP port 21 access from the internet'
```

File: query/vpc.sp [hunk 9]
```diff
@@ -357,4 +357,202 @@ query "vpc_ec2_transit_gateway_auto_accept_attachment_requests_disabled" {
...
+        )
+    )
+    select
+      type || ' ' || r.name as resource,
+      case
+        when g.name is null then 'ok'
+        else 'alarm'
+      end as status,
+      r.name || case
+        when g.name is null then ' restricts FTP data port 20 access from internet'
```
Comment (madhushreeray30) [inline query/vpc.sp:393]:
```suggestion
        when g.name is null then ' restricts FTP data port 20 access from the internet'
```

File: query/vpc.sp [hunk 10]
```diff
@@ -357,4 +357,202 @@ query "vpc_ec2_transit_gateway_auto_accept_attachment_requests_disabled" {
...
+        else ' allows FTP data port 20 access from internet'
```
Comment (madhushreeray30) [inline query/vpc.sp:394]:
```suggestion
        else ' allows FTP data port 20 access from the internet'
```

File: query/vpc.sp [hunk 11]
```diff
@@ -357,4 +357,202 @@ query "vpc_ec2_transit_gateway_auto_accept_attachment_requests_disabled" {
...
+        )
+    )
+    select
+      type || ' ' || r.name as resource,
+      case
+        when g.name is null then 'ok'
+        else 'alarm'
+      end as status,
+      r.name || case
+        when g.name is null then ' restricts SSH access from internet through port 22'
```
Comment (madhushreeray30) [inline query/vpc.sp:481]:
```suggestion
        when g.name is null then ' restricts SSH access from the internet through port 22'
```

File: query/vpc.sp [hunk 12]
```diff
@@ -357,4 +357,202 @@ query "vpc_ec2_transit_gateway_auto_accept_attachment_requests_disabled" {
...
+        else ' allows SSH access from internet through port 22'
```
Comment (madhushreeray30) [inline query/vpc.sp:482]:
```suggestion
        else ' allows SSH access from the internet through port 22'
```

File: query/vpc.sp [hunk 13]
```diff
@@ -357,4 +357,202 @@ query "vpc_ec2_transit_gateway_auto_accept_attachment_requests_disabled" {
...
+        )
+    )
+    select
+      type || ' ' || r.name as resource,
+      case
+        when g.name is null then 'ok'
+        else 'alarm'
+      end as status,
+      r.name || case
+        when g.name is null then ' restricts RDP access from internet through port 3389'
```
Comment (madhushreeray30) [inline query/vpc.sp:525]:
```suggestion
        when g.name is null then ' restricts RDP access from the internet through port 3389'
```

File: query/vpc.sp [hunk 14]
```diff
@@ -357,4 +357,202 @@ query "vpc_ec2_transit_gateway_auto_accept_attachment_requests_disabled" {
...
+        else ' allows RDP access from internet through port 3389'
```
Comment (madhushreeray30) [inline query/vpc.sp:526]:
```suggestion
        else ' allows RDP access from the internet through port 3389'
```

**Review by madhushreeray30 — COMMENTED** [review]
Please have a look at the comments, thanks!

**Review by madhushreeray30 — APPROVED** [review]
LGTM 👍

File: query/glue.sp [hunk 15]
```diff
@@ -59,4 +59,60 @@ query "glue_job_security_configuration_enabled" {
...
+      name || case
+        when (arguments -> 'encryption_configuration' -> 'cloudwatch_encryption' ->> 'cloudwatch_encryption_mode') = 'SSE-KMS'
+        and (arguments -> 'encryption_configuration' -> 'cloudwatch_encryption' ->> 'kms_key_arn') is not null
+        and (arguments -> 'encryption_configuration' -> 'job_bookmarks_encryption' ->> 'job_bookmarks_encryption_mode') = 'CSE-KMS'
+        and (arguments -> 'encryption_configuration' -> 'job_bookmarks_encryption' ->> 'kms_key_arn') is not null
+        and (arguments -> 'encryption_configuration' -> 's3_encryption' ->> 's3_encryption_mode') <> 'DISABLED'
+        and (arguments -> 'encryption_configuration' -> 's3_encryption' ->> 'kms_key_arn') is not null  then ' encryption enabled'
+        else ' encryption disabled'
+      end || '.' reason
+      --${local.common_dimensions_sql}
```
Comment (misraved) [inline query/glue.sp:112]: Why is this commented?

File: query/vpc.sp [hunk 16]
```diff
@@ -357,4 +357,202 @@ query "vpc_ec2_transit_gateway_auto_accept_attachment_requests_disabled" {
...
+      type || ' ' || name as resource,
+      case
+        when (arguments ->> 'egress') = 'true' then 'skip'
+        when (arguments ->> 'rule_action') = 'allow' and (arguments -> 'from_port') is null then 'alarm'
+        else 'ok'
+      end as status,
+      name || case
+        when (arguments ->> 'egress') = 'true' then ' is egress rule'
+        when (arguments ->> 'rule_action') = 'allow' and (arguments -> 'from_port') is null then ' allows access to all port'
+        else ' restricts access to all port'
```
Comment (misraved) [inline query/vpc.sp:550]:
```suggestion
        when (arguments ->> 'rule_action') = 'allow' and (arguments -> 'from_port') is null then ' allows access to all ports'
        else ' restricts access to all ports'
```

**Review by misraved — COMMENTED** [review]
@madhushreeray30 @Priyanka-Chatterjee-2000 please take a look at the review comments. Thanks!!

**Review by misraved — APPROVED** [review]

**Pull request merged**